drivers/include/adc_ng.h

```diff
@@ -0,0 +1,333 @@
+{
+    assert(adc < XFA_LEN(adc_ng_backend_t, adc_ng_backends));
+    const adc_ng_backend_t be = adc_ng_backends[adc];
+    return be.driver->mux(be.handle, chan);
+}
+
+/**
+ * @brief   Turn of the given ADC device
+ *
+ * @param[in]       adc     ADC device to turn off
+ */
+static inline void adc_ng_off(uint8_t adc)
+{
+    assert(adc < XFA_LEN(adc_ng_backend_t, adc_ng_backends));
+    const adc_ng_backend_t be = adc_ng_backends[adc];
+    if (be.driver->off) {
+        be.driver->off(be.handle);
+    }
+}
+
+/**
+ * @brief   Return the number of the highest single ended channel of the
+ *          given ADC
+ *
+ * @param[in]       adc     ADC to get the number of the highest single
+ *                          ended channel of
+ *
+ * @details All channel from zero to the returned value are single ended
+ *          channels
+ * @note    Some ADCs might have "holes" in the channel map, e.g. channels
+ *          0, 1 and 3 might be available, but channel 2 is not supported.
+ *          Checking for the return value `-ENXIO` in @ref adc_ng_init can be
+ *          used to detect these "holes".
+ *
+ * A single ended input is the "normal" ADC channel where a single input
+ * voltage is converted into an unsigned integer with the value 0 referring to
+ * an input voltage of 0 V and the highest value referring to an input voltage
+ * equal to the voltage reference (or higher).
+ */
+static inline uint8_t adc_ng_highest_single_ended_channel(uint8_t adc)
+{
+    assert(adc < XFA_LEN(adc_ng_backend_t, adc_ng_backends));
+    const adc_ng_backend_t be = adc_ng_backends[adc];
+    return be.driver->highest_single_ended_channel;
+}
+
+/**
+ * @name    Functions access to access the raw ADC output
+ *
+ * These functions return the binary value returned by the ADC, rather than
+ * meaningful physical units.
+ *
+ * @{
+ */
+
+/**
+ * @brief   Perform a single conversion using the specified ADC channel
+ *
+ * @param[in]       adc     ADC device to use
+ * @param[out]      dest    The result is stored here
+ *
+ * @pre     The ADC @p adc is initialized using @ref adc_ng_init
+ *
+ * @retval  0               Success
+ * @retval  <0              A device specific error occurred
+ *
+ * @note    The sample is returned as signed integer to allow for differential
+ *          two ended channels: If the input voltage on the positive input is
+ *          lower than the voltage at the negative input the result should be
+ *          negative. For "normal" single ended inputs the result must always
+ *          be positive or zero.
+ */
+static inline int adc_ng_single(uint8_t adc, int32_t *dest)
+{
+    assert(adc < XFA_LEN(adc_ng_backend_t, adc_ng_backends));
+    assert(dest);
+    const adc_ng_backend_t be = adc_ng_backends[adc];
+    return be.driver->single(be.handle, dest);
+}
+
+/**
+ * @brief   Initialize an ADC channel, perform a single conversion with
+ *          maximum resolution and range, and power it off again
```

Comment:
does not sound `quick` but simple or easy to me

Reply:
indeed, those names are better. Any preference?

Reply:
not really (just a little in favor of simple)

Reply:
i am also in favor of return values being return values and not error codes